### PR TITLE
Set default storage tier to first data tier

### LINKED ISSUE #594: Set default pool to first data pool for tiering
## Describe the bug
Currently, the default storage pool is set to system, but to behave similar to the standard policy file if there are any data pools other than system those should be the default. 

## How to Reproduce?
Please list the steps to help development teams reproduce the behavior

1. Have a storage system with multiple pools, some of which allow for dataOnly or dataAndMetadata.
2. List the policy file and the default SET POOL rule for CSI will be system instead of the expected pool

## Expected behavior
The default pool for file placement on files that don't match any SET POOL rules should default to the first data pool.

### PR DESCRIPTION
**Pull request by drolsonibm**
Instead of setting the default pool to system, we now behave similar
to the default policy settings. If there is a data pool other than
system it will default to the first one it finds.

## Pull request checklist

```release-notes

```

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

- Defaulting to system pool even if there are data pools that policy file typically would default to. This makes the behavior consistent

## What is the new behavior?

-

## How risky is this change?
- [ ] Small, isolated change
- [x] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Review by madhuthorat — APPROVED** [review]
LGTM

**Review by madhuthorat — APPROVED** [review]
LGTM

**Comment by Jainbrt** [comment]
@drolsonibm any target to get this PR merged in dev ?

File: driver/csiplugin/controllerserver.go [hunk 1]
```diff
@@ -764,13 +764,16 @@ func (cs *ScaleControllerServer) CreateVolume(ctx context.Context, req *csi.Crea
 		// then all files that do not match our rules will have no defined place to go. This sets a default rule with
 		// "lower" priority than the main policy as a catch all. If there is already a default rule in the main policy
 		// file then that will take precedence
-		// TODO: get pool info and find first non-system data pool and use that if possible, otherwise system
 		defaultPartitionName := "csi-defaultRule"
 		if !scaleVol.Connector.CheckIfDefaultPolicyPartitionExists(defaultPartitionName, scaleVol.VolBackendFs) {
```
[the remark this answers is not in the document]
Comment (deeghuge) [reply driver/csiplugin/controllerserver.go:776]: @Jainbrt Only check is happening always but that we cant avoid unless we keep local cache. If default policy exist then remaining part it skipped

**Review by deeghuge — APPROVED** [review]

**Pull request merged**